src/data/nat/factorial.lean

```diff
@@ -125,75 +137,237 @@ begin
   cases hn,
   { rw factorial_one,
     exact lt_factorial_self (succ_le_succ hi) },
-  { exact add_factorial_succ_lt_factorial_add_succ _ hi }
+  exact add_factorial_succ_lt_factorial_add_succ _ hi,
 end
 
 lemma add_factorial_succ_le_factorial_add_succ (i : ℕ) (n : ℕ) :
   i + (n + 1)! ≤ (i + (n + 1))! :=
 begin
   by_cases i2 : 2 ≤ i,
   { exact (n.add_factorial_succ_lt_factorial_add_succ i2).le },
-  { cases (not_le.mp i2) with _ i0,
-    { change 1 + (n + 1)! ≤ (1 + n + 1) * (1 + n)!,
-      rw [add_mul, one_mul, add_comm 1 n],
-      exact (add_le_add_iff_right _).mpr (one_le_mul (nat.le_add_left 1 n) (n + 1).factorial_pos) },
-    { rw [nat.le_zero_iff.mp (nat.succ_le_succ_iff.mp i0), zero_add, zero_add] } }
+  cases (not_le.mp i2) with _ i0,
+  { change 1 + (n + 1)! ≤ (1 + n + 1) * (1 + n)!,
+    rw [add_mul, one_mul, add_comm 1 n],
+    exact (add_le_add_iff_right _).mpr (one_le_mul (nat.le_add_left 1 n) (n + 1).factorial_pos) },
+  rw [nat.le_zero_iff.mp (nat.succ_le_succ_iff.mp i0), zero_add, zero_add],
 end
 
 lemma add_factorial_le_factorial_add (i : ℕ) {n : ℕ} (n1 : 1 ≤ n) :
   i + n! ≤ (i + n)! :=
 begin
   cases n1 with h,
   { exact self_le_factorial _ },
-  { exact add_factorial_succ_le_factorial_add_succ i h }
+  exact add_factorial_succ_le_factorial_add_succ i h,
 end
 
 end factorial
 
-section desc_fac
+/-! ### Ascending and descending factorials -/
 
-/-- desc_fac n k = (n + k)! / n! (as seen in `nat.desc_fac_eq_div`), but implemented recursively.
-This is closely related to `pochhammer`, but much less general. -/
-def desc_fac (n : ℕ) : ℕ → ℕ
+section asc_factorial
+
+/-- `n !+ k = (n + k)! / n!` (as seen in `nat !+_eq_div`), but implemented recursively
```

Comment:
What do you mean by `nat !+_eq_div`?

Reply:
Possibly an over-eager find+replace occurred?